stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -276,7 +276,7 @@ function cached_PATH_changed()
     global cached_PATH_string
     @lock(PATH_cache_lock, cached_PATH_string) !== get(ENV, "PATH", nothing)
 end
-const PATH_cache_finished = Base.Condition() # used for sync in tests
+const PATH_cache_finished = Threads.Condition() # used for sync in tests
```

Comment:
```suggestion
const PATH_cache_finished = Base.Event() # used for sync in tests
```

Looks like you aren't managing the state and lock during the tests, so this may need to be an Event in that case (which manages those for you) to avoid concurrency errors and missed events. The tests can call `Base.reset(PATH_cache_finished)` to prepare.

An alternate better option here might be making this a `Union{Task,Nothing}` so you can `wait()` on the worker task directly when required, e.g. so that tab completion with `<tab>` is still correct after that previous PR change.

Reply:
Updated to the latter